### DISCUSSION TITLE
Running a frontend script from a backend script

### DISCUSSION THREAD

**Discussion by r-nd-m** [TriliumNext/Trilium#4125] [Q&A] [answered]
Is there a way to run a frontend script from the backend?

I'm using labels in notes and some of the labels should contain only specific values. Since there is no predefined label list functionality I wanted to implement a script that runs on runOnAttributeCreate (which needs to be a backend script), checks whether input value for label is within a predefined dictionary and show an error if the value is outside the list of possible values - but I cannot find any way to run showError() nor run a frontend script from the backend script :(

**Comment by zadam** [comment, marked as answer]
Hi, this is conceptually "complicated" since there may be several frontends attached to one backend, each with several tabs, and it wouldn't be clear what should react to such invocations. The frontend -> backend direction OTOH is clear.

Having said that, there is a way to catch e.g. attribute creation on the frontend via `entityReloadedEvent` of a custom widget, an example is shown here: https://github.com/zadam/trilium/discussions/3926

**Comment by r-nd-m** [comment]
Thank you, gonna try it that way.
